server/src/main/java/org/opensearch/action/admin/cluster/state/ClusterStateRequest.java

```diff
@@ -211,4 +217,17 @@ public ClusterStateRequest waitForMetadataVersion(long waitForMetadataVersion) {
         this.waitForMetadataVersion = waitForMetadataVersion;
         return this;
     }
+
+    public void setIsCancellationTaskRequired(boolean isCancellationTaskRequired) {
```

Comment:
when will this be false? rename to shouldCancelOnTimeout

Reply:
This will be false for other admin actions where cancellation support is not given.

Reply:
should we move this to ```ClusterManagerNodeReadRequest```